### ISSUE TITLE
Allow new field option 'env' when mounting Secrets inside RUN commands

### ISSUE THREAD

**Issue by yannispgs** [hadolint/hadolint#1039]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

Dockerfile Syntax 1.10.0 has been released, allowing secrets to be mounted as environment variables (+ file) rather than as a file.
Instead of `target=<path>` we can use `env=<env_var_name>`.

As it is brand new, the current rules do not allow this new field and raise an error.

### Expected behavior

No error raised at all

### Actual behavior

```bash
hadolint Dockerfile

Dockerfile.test:6:36 unexpected "env=ENV_VAR ec" expecting "gid=", "id=", "mode=", "required", "required=False", "required=True", "required=false", "required=true", "uid=", source=, or target=
```

### Steps to reproduce the behavior

**Output of `hadolint --version`**

```bash
Haskell Dockerfile Linter 2.12.0
```

**Dockerfile**

```Dockerfile
# syntax=docker/dockerfile:1.10

FROM node:20

RUN --mount=type=secret,id=ENV_VAR,env=ENV_VAR echo "ENV_VAR=$ENV_VAR" > /tmp/env_var
```

Command run = `docker build . --secret 'id=ENV_VAR,env=ENV_VAR'`

**Comment by petski** [comment]
Reference to changelog for 1.10: https://docs.docker.com/build/buildkit/dockerfile-release-notes/#1100

Reference to build-secret-`env=VARIABLE`-feature: https://github.com/moby/buildkit/pull/5215

**Comment by ktzsolt** [comment]
Hi! Does anybody know a workaround for this?

**Comment by ktzsolt** [comment]
> > Hi! Does anybody know a workaround for this?
> 
> I think the only workaround would be to change your Dockerfile to work around the `env` statement with something like:
> 
> RUN  --mount=type=secret,id=my_key \
>     MY_KEY="$(cat /run/secrets/my_key)" \
>     export MY_KEY && \
>     ....
> 
> should give you an env var in a hadolint friendly way. Not sure how secure this is though.

Thanks, but I don't want to fiddle with the dockerfile that is working perfectly. 

TBH, I dropped hadolint, it looks like it is not actively maintained, switched to trivy instead with `trivy config Dockerfile` it works just fine, we already use it for CVE scanning in or CI anyways.

**Comment by lorenzo** [comment]
this will be released soon. The code changes are there already

**Comment by cschiewek** [comment]
@lorenzo Any update on when this will get released?

**Comment by yannispgs** [comment]
All my PRs checks have been failing for the last 3 months since it has not been released yet. @lorenzo a hint on when this will be released please ?

**Comment by rvillane** [comment]
This is an interesting situation:

1. In Github, the latest release is from 2022: https://github.com/hadolint/hadolint/releases/tag/v2.12.1-beta
2. In brew and other sources, the latest version is: 2.12.0
3. In DockerHub latest version is from 16 days ago: [v2.13.1-beta4-debian](https://hub.docker.com/layers/hadolint/hadolint/v2.13.1-beta4-debian/images/sha256-2478aca5b6fb881adcd9e645e2c697ec5ce5a1351fe1f07e4ca72cab43bb956c)

the latest hadolint version available is not consistent across all channels.

**Comment by ctrahey** [comment]
If someone can describe what the community can do to help straighten out the releases, we'd be happy to contribute a few hours.

**Comment by purajit** [comment]
(not a maintainer, just following the issue) they've fixed the issue with the releases and 2.13.1 is available everywhere. It looks like they need to now do a release of hadolint/language-docker and pull that into here. I've brought that up in https://github.com/hadolint/hadolint/issues/1092#issuecomment-3246096473
